### PR TITLE
New AWS module to manage RDS Clusters 

### PR DESCRIPTION
**Pull request by aioue**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

rds_cluster
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel e396d5d508) last updated 2016/08/24 12:13:10 (GMT +1000)
  lib/ansible/modules/core: (detached HEAD 368ca738fa) last updated 2016/08/24 12:13:21 (GMT +1000)
  lib/ansible/modules/extras: (detached HEAD 0749ce6faa) last updated 2016/08/24 12:13:30 (GMT +1000)
```
##### SUMMARY

This module manages the creation, modification, deletion, snapshot and restore of RDS Clusters for Aurora instances in AWS.

```
TASK [print new Aurora RDS cluster] ********************************************
ok: [localhost] => {
    "msg": {
        "changed": true, 
        "rds": {
            "DBClusters": [
                {
                    "AllocatedStorage": 1, 
                    "AvailabilityZones": [
                        "ap-southeast-2a", 
                        "ap-southeast-2b", 
                        "ap-southeast-2c"
                    ], 
                    "BackupRetentionPeriod": 10, 
                    "DBClusterIdentifier": "test-aurora-cluster", 
                    "DBClusterMembers": [], 
                    "DBClusterParameterGroup": "test-aurora-param-group", 
                    "DBSubnetGroup": "aag-vpc-99-sbg-rds-green", 
                    "DbClusterResourceId": "cluster-TQILXHYILLHWNEPZLLHWNEC5Q", 
                    "EarliestRestorableTime": "2016-08-28T23:24:54.858000+00:00", 
                    "Endpoint": "test-aurora-cluster.cluster-c41234a2bug.ap-southeast-2.rds.amazonaws.com", 
                    "Engine": "aurora", 
                    "EngineVersion": "5.6.10a", 
                    "HostedZoneId": "Z32T0VXXXEXS0V", 
                    "KmsKeyId": "arn:aws:kms:ap-southeast-2:486509402765:key/1b184ef4-1234-5678-a0f0-d5970a9f8e21", 
                    "LatestRestorableTime": "2016-08-28T23:24:54.858000+00:00", 
                    "MasterUsername": "testuser", 
                    "Port": 3306, 
                    "PreferredBackupWindow": "15:35-16:05", 
                    "PreferredMaintenanceWindow": "sun:18:34-sun:19:04", 
                    "Status": "available", 
                    "StorageEncrypted": true, 
                    "VpcSecurityGroups": [
                        {
                            "Status": "active", 
                            "VpcSecurityGroupId": "sg-d111c7b6"
                        }
                    ]
                }
            ], 
            "ResponseMetadata": {
                "HTTPStatusCode": 200, 
                "RequestId": "ab1de429-6d76-11e6-9d17-df1f331e0454"
            }
        }
    }
}
```

### CODE REVIEW

**Comment by aioue** [comment]
Migrated from https://github.com/ansible/ansible-modules-extras/pull/2804 by aioue (not original author)

**Comment by nitzmahone** [comment]
Looks like this one's using iteritems, which won't work under python3.

File: lib/ansible/modules/cloud/amazon/rds_cluster.py [hunk 1]
```diff
@@ -0,0 +1,843 @@
...
+        'replication_source_identifier': 'ReplicationSourceIdentifier',
+        'tags': 'Tags',
+        'storage_encrypted': 'StorageEncrypted',
+        'kms_key_id': 'KmsKeyId',
+        'skip_final_snapshot': 'SkipFinalSnapshot',
+        'final_db_snapshot_identifier': 'FinalDBSnapshotIdentifier',
+        'apply_immediately': 'ApplyImmediately',
+        'snapshot_identifer': 'SnapshotIdentifier'}
+
+    for (k, v) in camel_params.iteritems():
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/rds_cluster.py:758]: Should be `for k, v in camel_params.items()`

File: lib/ansible/modules/cloud/amazon/rds_cluster.py [hunk 2]
```diff
@@ -0,0 +1,843 @@
...
+                module.fail_json(msg="Parameter %s is not valid for %s command" % (k, command))
+        if k in required_vars:
+            if module.params.get(k):
+                params[v] = module.params[k]
+            else:
+                module.fail_json(msg="Parameter %s required for %s command" % (k, command))
+
+    tag_array = []
+    if module.params.get('tags'):
+        for tag, value in module.params.get('tags').iteritems():
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/rds_cluster.py:772]: Should be `for tag, value in module.params.get('tags').items()`

File: lib/ansible/modules/cloud/amazon/rds_cluster.py [hunk 3]
```diff
@@ -0,0 +1,843 @@
...
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: rds_cluster
+short_description: Manages RDS Clusters for Aurora instances
+description:
+  -Manages the creation, modification, deletion, snapshot and restore of RDS Clusters for Aurora instances in AWS
+version_added: "2.2"
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/rds_cluster.py:23]: Needs bumping to 2.3

File: lib/ansible/modules/cloud/amazon/rds_cluster.py [hunk 4]
```diff
@@ -0,0 +1,843 @@
...
+    default: null
+  engine:
+    description:
+      - The type of database.  Used only when command=create.
+    required: false
+    default: null
+    choices: ['aurora']
+  master_username:
+    description:
+      - Master database username. Used only when command=create.
```
Comment (ryansb) [inline lib/ansible/modules/cloud/amazon/rds_cluster.py:46]: Instead of command this module should use `state=` and present/absent.

**Review by ryansb — CHANGES_REQUESTED** [review]

**Comment by willthames** [comment]
duplicate_of #19782 

Although this PR is newer and has more feedback, PRs should likely be owned by the original author where possible so that they can update them.

**Comment by willthames** [comment]
duplicate_of #19782 

Although this PR is newer and has more feedback, PRs should likely be owned by the original author where possible so that they can update them.

**Comment by aioue** [comment]
Closing due to original author picking up the PR.

**Pull request closed without merging**